numismatic/cli.py

```diff
@@ -107,6 +139,7 @@ def list_all(state, feed, output):
     feed_client = Feed.factory(feed, cache_dir=state['cache_dir'],
                                requester=state['requester'])
     assets_list = feed_client.get_list()
+    # FIXME: This should a collector rather than write()
```

Comment:
What are the implications for not using a collector at these stages? It seems like it will only output to stdout?

Reply:
Yeah exactly. That's why it should be replaced with using collectors rather. What's preventing me from doing it is that the historic methods don't produce Events at the moment but rather their own json type objects. As soon as those methods are standardised to produce Events as well then we can make them fit into the collector paradigm properly.

Reply:
Perhaps, if we are thinking about modularizing things, what about making the collectors another module (like what I suggested for vizualization)

Reply:
I think the Collectors are a pretty core part of Numismatic. You could maybe make particular types like the SqlCollector or the DataFrameCollector optional plugins but as a whole they have to be part of Numismatic as they are not optional.